From 1da966ad75cc6015baeae7ddc113b5daecd2c382 Mon Sep 17 00:00:00 2001
From: David Malcolm <dmalcolm@redhat.com>
Date: Tue, 15 Oct 2024 16:24:40 -0400
Subject: [PATCH 8/8] FIXME: replace
 -fdiagnostics-format=sarif-file-2.2-prerelease with
 -fdiagnostics-set-output=sarif:version=2.2-prerelease

---
 gcc/common.opt                                           | 4 ----
 gcc/diagnostic.cc                                        | 9 ---------
 gcc/diagnostic.h                                         | 6 +-----
 .../gcc.dg/plugin/crash-test-ice-in-header-sarif-2.2.c   | 2 +-
 4 files changed, 2 insertions(+), 19 deletions(-)

diff --git a/gcc/common.opt b/gcc/common.opt
index 6a267b023d81..dceaa6e8dbf7 100644
--- a/gcc/common.opt
+++ b/gcc/common.opt
@@ -1495,10 +1495,6 @@ Enum(diagnostics_output_format) String(sarif-stderr) Value(DIAGNOSTICS_OUTPUT_FO
 EnumValue
 Enum(diagnostics_output_format) String(sarif-file) Value(DIAGNOSTICS_OUTPUT_FORMAT_SARIF_FILE)
 
-EnumValue
-Enum(diagnostics_output_format) String(sarif-file-2.2-prerelease) Value(DIAGNOSTICS_OUTPUT_FORMAT_SARIF_FILE_2_2_PRERELEASE)
-; undocumented
-
 fdiagnostics-parseable-fixits
 Common Var(flag_diagnostics_parseable_fixits)
 Print fix-it hints in machine-readable form.
diff --git a/gcc/diagnostic.cc b/gcc/diagnostic.cc
index c4c238e4c0c5..2242d87ba94c 100644
--- a/gcc/diagnostic.cc
+++ b/gcc/diagnostic.cc
@@ -1741,15 +1741,6 @@ diagnostic_output_format_init (diagnostic_context &context,
 						sarif_version::v2_1_0,
 						base_file_name);
       break;
-    case DIAGNOSTICS_OUTPUT_FORMAT_SARIF_FILE_2_2_PRERELEASE:
-      diagnostic_output_format_init_sarif_file
-	(context,
-	 line_table,
-	 main_input_filename_,
-	 json_formatting,
-	 sarif_version::v2_2_prerelease_2024_08_08,
-	 base_file_name);
-      break;
     }
 }
 
diff --git a/gcc/diagnostic.h b/gcc/diagnostic.h
index 47237b2adee0..27bff94050f7 100644
--- a/gcc/diagnostic.h
+++ b/gcc/diagnostic.h
@@ -84,11 +84,7 @@ enum diagnostics_output_format
   DIAGNOSTICS_OUTPUT_FORMAT_SARIF_STDERR,
 
   /* SARIF-based output, to a file.  */
-  DIAGNOSTICS_OUTPUT_FORMAT_SARIF_FILE,
-
-  /* Undocumented, for use by test suite.
-     SARIF-based output, to a file, using a prerelease of the 2.2 schema.  */
-  DIAGNOSTICS_OUTPUT_FORMAT_SARIF_FILE_2_2_PRERELEASE
+  DIAGNOSTICS_OUTPUT_FORMAT_SARIF_FILE
 };
 
 /* An enum for controlling how diagnostic_paths should be printed.  */
diff --git a/gcc/testsuite/gcc.dg/plugin/crash-test-ice-in-header-sarif-2.2.c b/gcc/testsuite/gcc.dg/plugin/crash-test-ice-in-header-sarif-2.2.c
index b5328060c394..1c4cfbe07239 100644
--- a/gcc/testsuite/gcc.dg/plugin/crash-test-ice-in-header-sarif-2.2.c
+++ b/gcc/testsuite/gcc.dg/plugin/crash-test-ice-in-header-sarif-2.2.c
@@ -1,7 +1,7 @@
 /* Test of an ICE triggered within a header file with SARIF 2.2  */
 
 /* { dg-do compile } */
-/* { dg-options "-fdiagnostics-format=sarif-file-2.2-prerelease" } */
+/* { dg-options "-fdiagnostics-set-output=sarif:version=2.2-prerelease" } */
 /* { dg-additional-options "-fno-report-bug" } */
 
 #include "crash-test-ice-in-header.h"  /* { dg-ice "" } */
-- 
2.26.3

